File: leetcode/1133. Largest Unique Number/1133.py
# https://helloacm.com/python-method-to-find-the-largest-unique-number-in-an-array/
# https://leetcode.com/problems/largest-unique-number/

class Solution:
    def largestUniqueNumber(self, A: List[int]) -> int:
        freq = collections.Counter(A)
        x = list(filter(lambda x: freq[x] == 1, A))
        return max(x) if len(x) > 0 else -1

File: leetcode/1133. Largest Unique Number/1133.js
// https://helloacm.com/how-to-find-the-largest-unique-number-in-array-using-javascript-functional-programming/
// https://leetcode.com/problems/largest-unique-number/

/**
 * @param {number[]} A
 * @return {number}
 */
var largestUniqueNumber = function(A) {
    const num = Math.max(...A.filter(a => A.indexOf(a) === A.lastIndexOf(a)));
    return isFinite(num) ? num : -1;
};
